server/src/controllers/patient_controller.ts

```diff
@@ -56,6 +56,28 @@ export class PatientController implements interfaces.Controller {
             res.json(error.statusCode || 500, { error: error.message });
         }
     }
+
+    @Get("/:patientId/statuses/fields", "extractJwtMiddleware", "isValidPatientMiddleware")
+    private async getPatientStatusFields(req: Request, res: Response): Promise<void> {
+        try {
+            const { value, error } = Joi.object({ patientId: Joi.number().required() })
+                .required()
+                .validate({ patientId: req.params.patientId });
+
+            const isSamePatient = req["token"].userId === value.patientId;
+
+            if (!isSamePatient || error) {
+                res.json(400, error);
+                return;
+            }
+
+            const StatusFields = await this.patientService.getPatientStatusFields(value.patientId);
```

Comment:
Nitpick: naming convention is camelCase so there is no conflict with types

```suggestion
            const statusFields = await this.patientService.getPatientStatusFields(value.patientId);
```